compressors/rANS.py

```diff
@@ -117,41 +117,32 @@ class rANSEncoder(DataEncoder):
     Detailed information in the overview
     """
 
-    def __init__(self, freqs: Frequencies, rans_params: rANSParams):
+    def __init__(self, rans_params: rANSParams):
         """init function
 
         Args:
             freqs (Frequencies): frequencies for which rANS encoder needs to be designed
             rans_params (rANSParams): global rANS hyperparameters
         """
-        self.freqs = freqs
         self.params = rans_params
 
     def rans_base_encode_step(self, s, state: int):
         """base rANS encode step
 
         updates the state based on the input symbols s, and returns the updated state
         """
-        f = self.freqs.frequency(s)
+        f = self.params.freqs.frequency(s)
         block_id = state // f
-        slot = self.freqs.cumulative_freq_dict[s] + (state % f)
-        next_state = block_id * self.freqs.total_freq + slot
+        slot = self.params.freqs.cumulative_freq_dict[s] + (state % f)
+        next_state = block_id * self.params.freqs.total_freq + slot
         return next_state
 
-    @cache
-    def max_shrunk_state_val(self, symbol):
-        """
-        max value the state can be before calling rans_base_encode_step function
-        """
-        f = self.freqs.frequency(symbol)
-        return self.params.RANGE_FACTOR * f * (1 << self.params.NUM_BITS_OUT) - 1
-
     def shrink_state(self, state: int, next_symbol) -> Tuple[int, BitArray]:
-        """stream out the lower bits of the state, until the state is below self.max_state_val(next_symbol)"""
+        """stream out the lower bits of the state, until the state is below params.max_shrunk_state[next_symbol]"""
         out_bits = BitArray("")
 
         # output bits to the stream to bring the state in the range for the next encoding
-        while state > self.max_shrunk_state_val(next_symbol):
+        while state > self.params.max_shrunk_state[next_symbol]:
```

Comment:
Is there a reason you don't store `(1 << self.params.NUM_BITS_OUT)` as a __post_init__ parameter in rANSParams instead of recalculating it each time?

Reply:
also `self.params.freqs.total_freq` can be defined in init

Reply:
Thanks ll make the change!

Reply:
@tpulkit, I see that we only use `1 << self.params.NUM_BITS_OUT` once in the rANS code.. so keeping it that way, as no point in introducing a variable if we are not using it multiple times

Reply:
defined `self.M = self.freqs.total_freq` in the `_post_init`